src/GraphQL/API/Enum.hs

```diff
@@ -50,6 +56,14 @@ instance forall conName f p b.
   genericEnumToValue (L1 _) =
     case nameFromSymbol @conName of
       Right name -> name
+      -- XXX: This is impossible to catch during validation, because we cannot
```

Comment:
I don't think this statement is true. When we run `genericEnumValues` we do the same name-check so all we need to do is make sure `genericEnumValues` runs for validation, which it does I think if we `deepseq` the value representation.